sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, find_cycle, is_directed_acyclic_graph
+from sql_cli.sql_directory_parser import SqlFile
+
+
+class DagCycle(Exception):
+    """An exception raised when DAG contains a cycle."""
```

Comment:
Should we have a separate file for custom exceptions?

Reply:
Makes sense. We could have more exceptions in the future.